cli/Dockerfile-bin

```diff
@@ -1,14 +1,14 @@
 ## compile binaries
-FROM gcr.io/runconduit/go-deps:dac3fae6 as golang
+FROM gcr.io/runconduit/go-deps:f39a1eed as golang
 ARG CONDUIT_VERSION
 WORKDIR /go/src/github.com/runconduit/conduit
 COPY cli cli
 COPY controller controller
 COPY pkg pkg
 RUN mkdir -p /out
-RUN CGO_ENABLED=0 GOOS=darwin go build -a -installsuffix cgo -o /out/conduit-darwin -ldflags "-X github.com/runconduit/conduit/pkg/version.Version=$CONDUIT_VERSION" ./cli
-RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o /out/conduit-linux -ldflags "-X github.com/runconduit/conduit/pkg/version.Version=$CONDUIT_VERSION" ./cli
-RUN CGO_ENABLED=0 GOOS=windows go build -a -installsuffix cgo -o /out/conduit-windows -ldflags "-X github.com/runconduit/conduit/pkg/version.Version=$CONDUIT_VERSION" ./cli
+RUN CGO_ENABLED=0 GOOS=darwin go build -race -a -installsuffix cgo -o /out/conduit-darwin -ldflags "-X github.com/runconduit/conduit/pkg/version.Version=$CONDUIT_VERSION" ./cli
+RUN CGO_ENABLED=0 GOOS=linux go build -race -a -installsuffix cgo -o /out/conduit-linux -ldflags "-X github.com/runconduit/conduit/pkg/version.Version=$CONDUIT_VERSION" ./cli
+RUN CGO_ENABLED=0 GOOS=windows go build -race -a -installsuffix cgo -o /out/conduit-windows -ldflags "-X github.com/runconduit/conduit/pkg/version.Version=$CONDUIT_VERSION" ./cli
```

Comment:
Just thinking out loud here -- I wonder how much additional time the `-race` flag adds to the build. Building this docker image is already pretty slow, since we produce three static binaries, one for each platform. I wouldn't want to slow it down more if race detection is expensive. Also, since we're building the same target with all three build commands, we'd get the same results if only 1 of the 3 built with race detection.

Reply:
per @pcalcado, i'm going to add race detection in a separate branch.

related to this specific Dockerfile, i'm only going to add `-race` back for ci and testing, per https://blog.golang.org/race-detector:
> race-enabled binaries can use ten times the CPU and memory, so it is impractical to enable the race detector all the time